### PR TITLE
feat: load price for currency and apply margin calculation in MarginSelectorForm

### PR DESCRIPTION
**Pull request by richard-ramos**

### CODE REVIEW

File: app/js/containers/Seller/4_SellerMarginContainer.jsx [hunk 1]
```diff
@@ -5,6 +5,10 @@ import {connect} from 'react-redux';
 import MarginSelectorForm from '../../components/Seller/MarginSelectorForm';
 import Loading from '../../components/ui/Loading';
 import newSeller from "../../features/newSeller";
+import network from '../../features/network';
+import prices from '../../features/prices';
+import {getPrices} from '../../features/prices/reducer';
```
Comment (alaibe) [inline app/js/containers/Seller/4_SellerMarginContainer.jsx]: You should not be able to access internal element of each feature, that should exposed through the index.
You can see example for others

File: app/js/containers/Seller/4_SellerMarginContainer.jsx [hunk 2]
```diff
@@ -26,6 +30,11 @@ class SellerMarginContainer extends Component {
     } else {
       this.setState({ready: true});
     }
+
+    this.props.fetchPrices({
+      from: [this._getSymbol()],
+      to: [this.props.seller.currency]
```
Comment (jrainville) [inline app/js/containers/Seller/4_SellerMarginContainer.jsx]: Why are those arrays?

File: app/js/containers/Seller/4_SellerMarginContainer.jsx [hunk 3]
```diff
@@ -26,6 +30,11 @@ class SellerMarginContainer extends Component {
     } else {
       this.setState({ready: true});
     }
+
+    this.props.fetchPrices({
+      from: [this._getSymbol()],
```
Comment (alaibe) [inline app/js/containers/Seller/4_SellerMarginContainer.jsx]: The problem with this solution is that the price changes all the time and if you left the page open for 1 hours the data displayed might not be accurate.

What about when we do the initialisation we also start a scheduler who is going to fetch prices for all token/currency periodically => e.g every 1 or n minutes.

We can then attach the prices to the different tokens we have

What do you think?

File: app/js/containers/Seller/4_SellerMarginContainer.jsx (hunk 2 above)
Comment (richard-ramos) [reply app/js/containers/Seller/4_SellerMarginContainer.jsx]: fetchPrice allows asking for the price for more than a single token symbol.  In this case _getSymbol() returns the seller selected asset, and currency, the fiat currency symbol (EUR, USD...)

File: app/js/containers/Seller/4_SellerMarginContainer.jsx [hunk 4]
```diff
@@ -48,13 +57,20 @@ class SellerMarginContainer extends Component {
     this.setState({marketType});
   };
 
+  _getSymbol(){
+    return Object.keys(this.props.tokens)
+                 .find(token => this.props.tokens[token].address === this.props.seller.asset);
```
Comment (alaibe) [inline app/js/containers/Seller/4_SellerMarginContainer.jsx]: you could do a selector for that I think

File: app/js/containers/Seller/4_SellerMarginContainer.jsx (hunk 3 above)
Comment (jrainville) [reply app/js/containers/Seller/4_SellerMarginContainer.jsx]: If we do that, don't forget to kill the interval on component unmount

Comment (alaibe) [reply app/js/containers/Seller/4_SellerMarginContainer.jsx]: I would not even do that :)
I would imagine the token always have a set of price attach to them that is up to date

So not related to the specific component but to the App

Comment (jrainville) [reply app/js/containers/Seller/4_SellerMarginContainer.jsx]: Ah ok, yeah, if the fetch is in the App, it's ok then

File: app/js/features/prices/saga.js [hunk 5]
```diff
@@ -16,4 +18,23 @@ export function *onFetchPrices() {
   yield takeEvery(FETCH_PRICES, doFetchPrices);
 }
 
-export default [fork(onFetchPrices)];
+
+function *fetchPricesPeriodically() {
```
Comment (jrainville) [inline app/js/features/prices/saga.js]: No need to put "periodically" in the function name, since the function itself doesn't do an interval

**Pull request merged**